pac4j-core/src/main/java/org/pac4j/core/profile/ProfileManager.java

```diff
@@ -110,6 +110,8 @@ protected void removeOrRenewExpiredProfiles(final LinkedHashMap<String, U> profi
                         final Optional<U> newProfile = client.get().renewUserProfile(profile, context);
                         if (newProfile.isPresent()) {
                             profiles.put(key, newProfile.get());
+                            // TODO GET BOOLEANS FROM CONFIG OR MOVE THIS LOGIC ELSEWHERE
+                            save(true, newProfile.get(), false);
```

Comment:
You raise a good point here: it's a bug. We should have removed the profile when expired.

It's fixed now: https://github.com/pac4j/pac4j/commit/30236ef4d230520d93325df98c44407b9a1bd5c0

Can you remove this change?

Reply:
Thank you for your swift response. That was exactly what was needed.

Reply:
I tested it by running it in a play framework project, which uses the KeycloakOidcClient.

I set org.pac4j package logging level to trace, then i set the access tokens lifespan to 1 minute in my keycloak instance, and saw the profile getting refreshed over and over again.

I also verified that the profile is removed once both the access token and refresh token has expired.

Reply:
Excellent!